src/rosbag_analysis_tools/scripts/IHP2_analysis_scripts/IHP2_Analysis_Script.ipynb

```diff
@@ -0,0 +1,534 @@
+    "#Input file path where output graphs should be saved to, ending with \"/\"\n",
+    "file_save = \"\" #Input here\n",
+    "\n",
+    "#Input vehicle names for formatting graphs\n",
+    "lead_v1_name = \"\" #Input here\n",
+    "follower_v2_name = \"\" #Input here\n",
+    "follower_v3_name = \"\" #Input here\n",
+    "test_number = \"\" #Input test number in format \"T##\"\n",
+    "run_number = \"\" #Input run number in format \"R#\"\n",
+    "\n",
+    "if lead_v1_file_read != \"\":\n",
+    "    lead_v1_bag = rosbag.Bag(file_read_location + lead_v1_file_read)\n",
+    "if follower_v2_file_read != \"\":\n",
+    "    follower_v2_bag = rosbag.Bag(file_read_location + follower_v2_file_read)\n",
+    "if follower_v3_file_read != \"\":\n",
+    "    follower_v3_bag= rosbag.Bag(file_read_location + follower_v3_file_read) \n",
+    "if lead_v1_file_read != \"\" and follower_v2_file_read != \"\" and follower_v3_file_read != \"\" :\n",
+    "    three_vehicle_bool = True"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Use this section for 2 vehicle test run\n",
+    "two_vehicle_bool = False\n",
+    "\n",
+    "#Input the file paths to the two bag files\n",
+    "#lead_v1b is the leader vehicle (vehicle 1)\n",
+    "#follower_v2b is the first follower vehicle (vehicle 2)\n",
+    "\n",
+    "file_b_read_location = \"\" #Input folder path where bag files are stored, ending with \"/\"\n",
+    "lead_v1b_file_read = \"\" #Input file name here\n",
+    "follower_v2b_file_read = \"\" #Input file name here\n",
+    "\n",
+    "#Input file path where output graphs should be saved to, ending with \"/\"\n",
+    "file_b_save = \"\" #Input here\n",
+    "\n",
+    "#Input vehicle names for formatting graphs\n",
+    "lead_v1b_name = \"\" #Input here\n",
+    "follower_v2b_name = \"\" #Input here\n",
+    "test_number_b = \"\" #Input test number in format \"T##\"\n",
+    "run_number_b = \"\" #Input run number in format \"R#\"\n",
+    "\n",
+    "if lead_v1b_file_read != \"\":\n",
+    "    lead_v1b_bag = rosbag.Bag(file_b_read_location + lead_v1b_file_read)\n",
+    "if follower_v2b_file_read != \"\":\n",
+    "    follower_v2b_bag = rosbag.Bag(file_b_read_location + follower_v2b_file_read)\n",
+    "if lead_v1b_file_read != \"\" and follower_v2b_file_read != \"\":\n",
+    "    two_vehicle_bool = True"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# DO NOT CHANGE ANYTHING BELOW THIS LINE"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#3 vehicle test run\n",
+    "if three_vehicle_bool == True :\n",
+    "\n",
+    "      # delay time until start carma engaged\n",
+    "      lead_v1_engaged_flag = 0\n",
+    "      for topic, msg, t in lead_v1_bag.read_messages(topics = [\"/guidance/state\"]):\n",
+    "            if msg.state == 4:\n",
```

Comment:
Consider adding a comment stating that '4' indicates that the CARMA Platform system is engaged